rest-api/dao/metrics_cache_dao.py

```diff
@@ -311,50 +357,287 @@ def to_client_json(self, result_set):
+                                     CASE WHEN q.code_id = {2} THEN 1 ELSE 0 END   AS WhatRaceEthnicity_Black,
+                                     CASE WHEN q.code_id = {3} THEN 1 ELSE 0 END   AS WhatRaceEthnicity_White,
+                                     CASE WHEN q.code_id = {4} THEN 1 ELSE 0 END   AS WhatRaceEthnicity_AIAN,
+                                     CASE WHEN q.code_id IS NULL THEN 1 ELSE 0 END AS UNSET,
+                                     CASE WHEN q.code_id = {5} THEN 1 ELSE 0 END   AS WhatRaceEthnicity_RaceEthnicityNoneOfThese,
+                                     CASE WHEN q.code_id = {6} THEN 1 ELSE 0 END   AS WhatRaceEthnicity_Asian,
+                                     CASE WHEN q.code_id = {7} THEN 1 ELSE 0 END   AS PMI_PreferNotToAnswer,
+                                     CASE WHEN q.code_id = {8} THEN 1 ELSE 0 END   AS WhatRaceEthnicity_MENA,
+                                     CASE WHEN q.code_id = {9} THEN 1 ELSE 0 END   AS PMI_Skip,
+                                     CASE WHEN q.code_id = {10} THEN 1 ELSE 0 END   AS WhatRaceEthnicity_NHPI
+                              FROM participant p
+                                     INNER JOIN participant_summary ps ON p.participant_id = ps.participant_id
+                                     LEFT JOIN
+                                   (
+                                     SELECT qr.participant_id, qra.value_code_id as code_id
+                                     FROM questionnaire_question qq,
+                                          questionnaire_response_answer qra,
+                                          questionnaire_response qr
+                                     WHERE qq.questionnaire_question_id = qra.question_id
+                                       AND qq.code_id = {0}
+                                       AND qra.questionnaire_response_id = qr.questionnaire_response_id
+                                   ) q ON p.participant_id = q.participant_id
+                              WHERE p.hpo_id=:hpo_id AND p.hpo_id <> :test_hpo_id
+                                AND (ps.email IS NULL OR NOT ps.email LIKE :test_email_pattern)
+                                AND p.withdrawal_status = :not_withdraw
+                                AND p.is_ghost_id IS NOT TRUE
+                                AND ps.questionnaire_on_the_basics = 1
+                            ) x
+                       GROUP BY participant_id, hpo_id, sign_up_time
+                     ) p,
+                     calendar,
+                     hpo
+                WHERE p.hpo_id = hpo.hpo_id
+                  AND calendar.day >= :start_date
+                  AND calendar.day <= :end_date
+                  AND calendar.day >= Date(p.sign_up_time)
+              ) y
+              GROUP BY day, hpo_id, name
+              ;
+        """.format(self.race_code_dict['Race_WhatRaceEthnicity'],
```

Comment:
This is not sql injection safe, can you parameterize it like we do in other places ?

Reply:
Done